backend/data.go

```diff
@@ -37,11 +41,45 @@ func (fn QueryDataHandlerFunc) QueryData(ctx context.Context, req *QueryDataRequ
 // QueryDataRequest contains a single request which contains multiple queries.
 // It is the input type for a QueryData call.
 type QueryDataRequest struct {
+	ForwardHTTPHeaders
 	PluginContext PluginContext
 	Headers       map[string]string
 	Queries       []DataQuery
 }
 
+func (req *QueryDataRequest) SetHTTPHeader(key, value string) {
+	if req.Headers == nil {
+		req.Headers = map[string]string{}
+	}
+
+	req.Headers[fmt.Sprintf("http_%s", key)] = value
+}
+
+func (req QueryDataRequest) GetHTTPHeader(key string) string {
```

Comment:
This method `GetHTTPHeader` and also `GetHTTPHeaders` have a value receiver, but `SetHTTPHeader` has a pointer receiver instead. Having a mix of pointer/value receivers can be confusing. Maybe there's a reason why then `Get` ones have a value receiver that I am not seeing, but I would consider switching all of them to pointer receivers if possible

Reply:
My idea is that they shouldn't change anything on the request object, hence no pointer receiver. Not sure if switching all of them make sense?

Reply:
Good point. I'll change

Reply:
Please review again thanks